### PR TITLE
[tune] hyperopt searcher to support tune.choice([[1,2],[3,4]]).

### LINKED ISSUE #24050: [Tune]Tune convert search space incorrectly into HyperOpt search space
### What happened + What you expected to happen

If passing nested list as search space to `tune.choice()`, the hyperparameter being sampled is an empty dictionary instead.
For example,
```python
config={'layers': tune.choice([[200, 300], [300, 400]])}  # layers would be {} when passed to trainable when using HyperOpt searcher
```

### Versions / Dependencies

ray=1.11.0
python=3.7.12
OS:unix

### Reproduction script

```python
from ray import tune
from ray.tune.suggest.hyperopt import HyperOptSearch

from ray import tune


def objective(x, a, b):  
    return a * (x ** 0.5) + b


def trainable(config):  
    print(config['layers']  # you only get empty dictionary: {}
    for x in range(20):  
        score = objective(x, config["a"], config["b"])

        tune.report(score=score) 

search_space = {"a": tune.randint(0, 20), "b": tune.randint(0, 40), 'layers': tune.choice([[300], [200, 100]])},
searcher = HyperOptSearch( metric='score', mode='max')
tune.run(
    trainable,
    config=search_space,
    metric="score",
    mode="min",
    search_alg=searcher,
    stop={"training_iteration": 20},
)
```

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by xwjiang2010**
## Why are these changes needed?

Have Hyperopt Searcher to support tune.choice([1,2],[3,4]) type search space.

## Related issue number

Closes #24050

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/tests/test_sample.py [hunk 1]
```diff
@@ -982,6 +982,30 @@ def testConvertHyperOpt(self):
...
+            space=converted_config, random_state_seed=1234, metric="a", mode="max"
+        )
+        searcher2 = HyperOptSearch(
+            space=hyperopt_config, random_state_seed=1234, metric="a", mode="max"
+        )
+
+        config1 = searcher1.suggest("0")
+        config2 = searcher2.suggest("0")
+
+        self.assertEqual(config1, config2)
```
Comment (Yard1) [inline python/ray/tune/tests/test_sample.py:1007]: Should we also check in the test that the suggestions are either of the two lists?

**Review by Yard1 — APPROVED** [review]
Looks good, one nit.

**Review by krfricke — COMMENTED** [review]
Looks good, let's add the simple assertion Antoni suggested. Also, can you provide a short PR description so I can use it as the commit message?

File: python/ray/tune/tests/test_sample.py [hunk 2]
```diff
@@ -982,6 +982,32 @@ def testConvertHyperOpt(self):
...
+
+        self.assertTrue(config1.get("a") in [(1, 2), (3, 4)])
```
Comment (krfricke) [inline python/ray/tune/tests/test_sample.py:1009]: Just to clarify, this means that hyperopt converts into tuples even though we specify lists, right?

Comment (xwjiang2010) [reply python/ray/tune/tests/test_sample.py:1009]: Yes, correct.

Comment (Yard1) [reply python/ray/tune/tests/test_sample.py:1009]: Let's add a comment for it!

Comment (xwjiang2010) [reply python/ray/tune/tests/test_sample.py:1009]: That's a good idea. Added it.

**Pull request merged**